app/workers/sync-In.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+require_once __DIR__ . '/../worker.php';
+
+use Appwrite\Extend\Exception;
+use Utopia\App;
+use Utopia\Cache\Cache;
+use Utopia\CLI\Console;
+use Utopia\Database\DateTime;
+use Utopia\Queue;
+use Utopia\Queue\Message;
+
+if (App::getEnv('_APP_REGION', 'default') === 'default') {
+    throw new Exception(Exception::GENERAL_SERVER_ERROR);
+}
+
+global $register;
+global $dsn;
+
+$connection = new Queue\Connection\Redis($dsn->getHost(), $dsn->getPort());
+$adapter    = new Queue\Adapter\Swoole($connection, 2, 'syncIn');
+$server     = new Queue\Server($adapter);
+
+$server->job()
+    ->inject('message')
+    ->inject('cache')
+    ->action(function (Message $message, Cache $cache) {
+        $time = DateTime::now();
+        $payload = $message->getPayload()['value'];
+        $cache->setDisableListeners(true);
+        foreach ($payload['keys'] ?? [] as $key) {
+            Console::info("[{$time}] Purging  {$key}");
+            $cache->purge($key);
+        }
+        $cache->setDisableListeners(false);
+    });
+
+$server
+    ->error()
+    ->inject('error')
+    ->inject('logError')
```

Comment:
We should call this logger, and use the logger to log the error, we would usually avoid saving functions or primitives as dependencies.